ElementInternalsType/explainer.md

```diff
@@ -11,23 +11,257 @@
+
+```js
+class CustomButton extends HTMLElement {
+    static buttonActivationBehaviors = true;
+}
+customElements.define('custom-button', CustomButton);
+```
+
+**Supported attributes:**
+
+When `static buttonActivationBehaviors = true` is set, the custom element would gain support for button activation-specific attributes:
+
+- `commandfor` - Targets another element to be invoked
+- `command` - Indicates to the targeted element which action to take
+
+**Supported properties:**
+
+The `ElementInternals` interface would be extended with button activation-specific properties:
+
+- `commandForElement` - reflects the `commandfor` attribute
+- `command` - reflects the `command` attribute
+
+**Supported events:**
+
+- `command` event - Fired on the element referenced by `commandfor`
+- `click` event - Fired on the custom element
+
+**IDL definitions:**
+```webidl
+partial interface ElementInternals {
+  attribute Element? commandForElement;
+  attribute DOMString command;
+};
+```
+
+**Implicit behaviors:**
+Beyond attributes, properties, and events, custom elements with `buttonActivationBehaviors = true` also gain native behaviors related to button activation:
+- **Form association**: The custom element automatically becomes form-associated
+- **Click event activation**: Fire click events when activated via mouse click, Enter key, Space key, or other activation methods
+- **Focusable by default**: The element becomes focusable and participates in tab navigation without requiring `tabindex`
+- **Default ARIA semantics**: Have an [button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/button_role) default ARIA role. Note ElementInternals.role is not changed. This is following the pattern where Element.role does not reflect native `<button>`'s `button` ARIA role.
+
+### Order of precedence regarding ARIA role
+The order is `<custom-button role=foo>` > `ElementInternals.role` > default `button` role  via `buttonActivationBehaviors`
+
+When `buttonActivationBehaviors = true`, the custom element's default ARIA role will become `button` and this will be the used role if no explicit role is specified by the author. If the author sets `elementInternals.role`, the value of `elementInternals.role` will be the used role, taking precedence over the default role. If the author sets the `role` attribute on the custom element, the value of the `role` attribute will be the used role, taking precedence over both `elementInternals.role` and the default role.

```

Comment:
I think only either this paragraph or the previous sentence is necessary; together they seem redundant.